README.md

```diff
@@ -129,6 +129,80 @@ file { '/etc/tmpfiles.d/foo.conf':
 ~> Class['systemd::tmpfiles']
 ```
 
+### timer units
+Create a timer unit and a service unit to execute from
+that timer at the same time. The following will manage timer unit 
+service and enable it.
+
+```puppet
+systemd::timer{'runoften.timer':
+  timer_source   => "puppet:///modules/${module_name}/runoften.timer",
+  service_source => "puppet:///modules/${module_name}/runoften.service",
+  active         => true,
+  enable         => true,
+}
+```
+or a trivial daily run and manage the timer unit manually
+
+```puppet
+$_timer = @(EOT)
+[Timer]
+OnCalendar=daily
+RandomizedDelaySec=1d
+EOT
+
+$_service = @(EOT)
+[Service]
+Type=oneshot
+ExecStart=/usr/bin/touch /tmp/file
+EOT
+
+if versioncmp($facts['puppetversion'],'6.1.0') < 0 {
+  include systemd::systemctl::daemon_reload
+  Systemd::Timer['daily.timer'] ~> Class[Systemd::Systemctl::Daemon_reload]
```

Comment:
This relationship already exists??
Should be
```
Class[Systemd::Systemctl::Daemon_reload] -> Service['daily.timer']
```
???

Reply:
Seems bad to notify things inside the unit_file type , assumes it's contents.

Reply:
Not sure I follow. https://github.com/camptocamp/puppet-systemd/blob/f81339ad10516905e460d2899432ce92d83a35d2/manifests/unit_file.pp#L86 is the relationship I think you're duplicating.

Reply:
Actually, I think it's
```
Systemd::Timer['daily.timer'] ~> Service['daily.timer']
Class[Systemd::Systemctl::Daemon_reload] ~> Service['daily.timer']
```
that would be needed.

Perhaps systemd::timer can always make sure a service resource is declared itself? (which I think would be ok for timer units created with this module)
```puppet
unless $enable or $active {
  # if either $enable or $active are set, unit_file will create a service resource and 
  service { $name:
    provider  => systemd,
    subscribe => [
      Class[Systemd::Systemctl::Daemon_reload],
      Systemd::Unit_file[$name],
    ],
  }
}
```
or
```puppet
service { $name:
  provider  => systemd,
  enable    => $enable,
  active    => $active,
  subscribe => [
    Class[Systemd::Systemctl::Daemon_reload],
    Systemd::Unit_file[$name],
  ],
}
systemd::unit_file{$name:
    # as before but active/enable parameters removed
    ensure    => $ensure,
    content   => $timer_content,
    source    => $timer_source,
    path      => $path,
    owner     => $owner,
    group     => $group,
    mode      => $mode,
    show_diff => $show_diff,
}
```

Reply:
> Not sure I follow.
was me confused.

Creating the service inside timer.pp seems a shame to not reuse unit_file code.